bot.py

```diff
@@ -99,6 +110,131 @@ async def get_last_match(ctx, name: str):
     os.remove("df_styled.png")
 
 
+@bot.command(name="add", help="Add the players to the list")
+async def add_summoner(ctx, *, message):
+    """Writes list of summoners to local
+    json file and sends the list to the bot"""
+
+    async with ctx.typing():
+        await asyncio.sleep(1)
+
+    json_path = "data/data.json"
+
+    # create a directory containing json file to store data for added summoners
+    if not os.path.exists(json_path):
+        os.makedirs("data")
+        with open(json_path, "w"):
+            pass
+
+    # accepting comma, space and comma plus space
+    players_list = [x.strip() for x in message.split(",")]
+    players_list = list(filter(None, players_list))
+
+    file_data = ""
+
+    # initializing number of players to display
+    number_of_players = 0
+
+    # initializing server id to a variable
+    server_id = str(ctx.guild.id)
+
+    # storing the json file into a variable
+    if os.path.getsize(json_path) > 0:
+        with open(json_path, "r") as file:
+            file_data = json.load(file)
+
+        # if server id exist in json, add number of players
+        if server_id in file_data:
+            number_of_players += len(file_data[server_id])
+
+    # send error message to bot then exits out of the function if an error with summoner's name
+    for count, _ in enumerate(players_list):
+        if not check_summoner_name(players_list[count]):
```

Comment:
Again, this is calling API even if user command includes already added summoner name.
You should first do;
1) Check if given summoner name exists in `data` fiel
2) Check if we have enough info like tier and rank info in our data

Reply:
Also, what if we already have 10 players in our data?